pkg/provisioners/channel_util.go

```diff
@@ -63,19 +63,35 @@ func RemoveFinalizer(o metav1.Object, finalizerName string) {
 	o.SetFinalizers(finalizers.List())
 }
 
-func CreateK8sService(ctx context.Context, client runtimeClient.Client, c *eventingv1alpha1.Channel) (*corev1.Service, error) {
+type k8sServiceOption func(*corev1.Service) error
+
+// ExternalService is a functional option for CreateK8sService to create a K8s service of type ExternalName
+func ExternalService(c *eventingv1alpha1.Channel) k8sServiceOption {
+	return func(svc *corev1.Service) error {
+		svc.Spec = corev1.ServiceSpec{
+			Type:         corev1.ServiceTypeExternalName,
+			ExternalName: names.ServiceHostName(channelDispatcherServiceName(c.Spec.Provisioner.Name), system.Namespace()),
+		}
+		return nil
+	}
+}
+
+func CreateK8sService(ctx context.Context, client runtimeClient.Client, c *eventingv1alpha1.Channel, opts ...k8sServiceOption) (*corev1.Service, error) {
```

Comment:
Doesn't `k8sServiceOption` need to be public for it to be an argument of a public method?

How about using `ServiceOption` instead?

Reply:
It doesn't have to be public.
The reason I kept in private is so that the concrete implementations of options could be written in channel_util and control the options that New provides. Example: ExternalService is the only option that is provided for now.